pymc3/tests/test_distributions.py

```diff
@@ -1505,6 +1521,156 @@ def test_batch_multinomial(self):
+        with Model() as model:
+            m = DirichletMultinomial("m", n=n, alpha=alpha, shape=(2, 4))
+        assert_allclose(m.distribution.mode.eval().sum(axis=-1), n)
+
+    def test_dirichlet_multinomial_vec(self):
+        vals = np.array([[2, 4, 4], [3, 3, 4]])
+        alpha = np.array([0.2, 0.3, 0.5])
+        n = 10
+
+        with Model() as model_single:
+            DirichletMultinomial("m", n=n, alpha=alpha, shape=len(alpha))
+
+        with Model() as model_many:
+            DirichletMultinomial("m", n=n, alpha=alpha, shape=vals.shape)
+
+        assert_almost_equal(
+            np.asarray([dirichlet_multinomial_logpmf(v, n, alpha) for v in vals]),
+            np.asarray([model_single.fastlogp({"m": val}) for val in vals]),
+            decimal=4,
+        )
+
+        assert_almost_equal(
+            np.asarray([dirichlet_multinomial_logpmf(v, n, alpha) for v in vals]),
+            model_many.free_RVs[0].logp_elemwise({"m": vals}).squeeze(),
+            decimal=4,
+        )
+
+        assert_almost_equal(
+            sum([model_single.fastlogp({"m": val}) for val in vals]),
+            model_many.fastlogp({"m": vals}),
+            decimal=4,
+        )
+
+    def test_dirichlet_multinomial_vec_1d_n(self):
+        vals = np.array([[2, 4, 4], [4, 3, 4]])
+        alpha = np.array([0.2, 0.3, 0.5])
+        ns = np.array([10, 11])
+
+        with Model() as model:
+            DirichletMultinomial("m", n=ns, alpha=alpha, shape=vals.shape)
+
+        assert_almost_equal(
+            sum([dirichlet_multinomial_logpmf(val, n, alpha) for val, n in zip(vals, ns)]),
+            model.fastlogp({"m": vals}),
+            decimal=4,
+        )
+
+    def test_dirichlet_multinomial_vec_1d_n_2d_alpha(self):
+        vals = np.array([[2, 4, 4], [4, 3, 4]])
+        alphas = np.array([[0.2, 0.3, 0.5], [0.9, 0.09, 0.01]])
+        ns = np.array([10, 11])
+
+        with Model() as model:
+            DirichletMultinomial("m", n=ns, alpha=alphas, shape=vals.shape)
+
+        assert_almost_equal(
+            sum([dirichlet_multinomial_logpmf(val, n, alpha) for val, n, alpha in zip(vals, ns, alphas)]),
+            model.fastlogp({"m": vals}),
+            decimal=4,
+        )
+
+    def test_dirichlet_multinomial_vec_2d_alpha(self):
+        vals = np.array([[2, 4, 4], [3, 3, 4]])
+        alphas = np.array([[0.2, 0.3, 0.5], [0.3, 0.3, 0.4]])
+        n = 10
+
+        with Model() as model:
+            DirichletMultinomial("m", n=n, alpha=alphas, shape=vals.shape)
+
+        assert_almost_equal(
+            sum([dirichlet_multinomial_logpmf(val, n, alpha) for val, alpha in zip(vals, alphas)]),
+            model.fastlogp({"m": vals}),
+            decimal=4,
+        )
+
+    def test_batch_dirichlet_multinomial(self):
+        n = 10
+        vals = np.zeros((4, 5, 3), dtype="int32")
+        alpha = np.zeros_like(vals, dtype=theano.config.floatX)
```

Comment:
> Are you sure you are passing valid alpha parameters to the DM copy of the test (i.e., not including any zero)?

Ah, there are indeed 0s in the alpha value being passed!

Reply:
The test should still fail, since it is comparing the DM.logp with the Multinomial.logp, and the samples are also likely to not match with what would be expected from the Multinomial... 

@lucianopaz, as @bsmith89  mentioned above, do you think a test more in line with the `test_multinomial_vec_1d_n...` would suffice to test a `3d_n` is being properly handled? https://github.com/pymc-devs/pymc3/blob/2824027837931a6c2f1d4ac5f210eb36b8fdf7a7/pymc3/tests/test_distributions.py#L1677-L1689